dashboard/src/components/TestDetails/TestDetails.tsx

```diff
@@ -22,22 +24,25 @@ import IssueSection from '../Issue/IssueSection';
 const emptyValue = '-';
 const valueOrEmpty = (value: string | undefined): string => value || emptyValue;
 
-const TestDetailsSection = ({
-  test,
-  context,
-}: {
-  test: TTestDetails;
-  context: string;
-}): JSX.Element => {
+const TestDetailsSection = ({ test }: { test: TTestDetails }): JSX.Element => {
   const intl = useIntl();
+  const historyState = useRouterState({ select: s => s.location.state });
   const hardware: string =
     test.environment_compatible?.join(' | ') ??
     intl.formatMessage({ id: 'global.unknown' });
 
-  const buildDetailsLink =
-    `${window.location.origin}` +
-    `/${context}/${test.git_commit_hash}/build/${test.build_id}` +
-    `${window.location.search}`;
+  let buildDetailsLink = '';
+  if (historyState.from !== undefined) {
+    buildDetailsLink =
+      `${window.location.origin}` +
+      `/${historyState.from}/${historyState.id}/build/${test.build_id}` +
+      `${window.location.search}`;
```

Comment:
use tnastack router for the search params

Reply:
Since I need the search params string in this case (the way the component is built it requires a link string to be passed to it) isn't location.search acceptable? As far as I searched I didn't find a way for tanstack router to return me the search params string. If location.search is not acceptable, a solution would be to implement a `stringify` function to get the search params string from the object returned by `useSearch` or use a library like [query-string](https://github.com/sindresorhus/query-string) that seems to do the job

Reply:
I'm seeing the root cause, we had unnecessary `window.location`  here, we don't need that, build details and test details are internal urls, they can be accesses with `/` 
and aside from that we can just use regular tanstack links with the callback search params